lib/Parser/Parse.cpp

```diff
@@ -10682,7 +10682,7 @@ bool Parser::CheckAsmjsModeStrPid(IdentPtr pid)
         !m_pscan->IsEscapeOnLastTkStrCon() &&
         wcsncmp(pid->Psz(), L"use asm", 10) == 0);
 
-    if (isAsmCandidate && m_scriptContext->IsInDebugMode())
+    if (isAsmCandidate && m_scriptContext->IsScriptContextInDebugMode())
```

Comment:
So if we have asmjs code in the library file we will emit this error to the debugger. (we were doing it before your change as well). In the future we should not raise this message when the current code is library code - for the user it does not make sense.